scripts/create_test_subset.py

```diff
@@ -187,15 +192,67 @@ def main(
             if analysis:
                 logger.info(f'Processing {a_type} analysis entry')
                 am = AnalysisModel(
-                    type=a_type,
+                    type=AnalysisType(a_type),
                     output=_copy_files_in_dict(analysis['output'], project),
-                    status=analysis['status'],
+                    status=AnalysisStatus(analysis['status']),
                     sample_ids=[s['id']],
                 )
                 logger.info(f'Creating {a_type} analysis entry in test')
                 aapi.create_new_analysis(project=target_project, analysis_model=am)
         logger.info(f'-')
 
+    sample_external_ids = [sample['external_id'] for sample in samples]
+    papi.fill_in_missing_participants(target_project)
+    participant_map = papi.get_participant_id_map_by_external_ids(
+        project, sample_external_ids
+    )
+
+    participant_ids = list(participant_map.values())
+    family_ids = transfer_families(project, target_project, participant_ids)
+    transfer_ped(project, target_project, family_ids)
+
+
+def transfer_families(initial_project, target_project, participant_ids) -> List[str]:
+    """Pull relevant families from the input project, and copy to target_project"""
+    families = fapi.get_families(
+        project=initial_project,
+        participant_ids=participant_ids,
+    )
+
+    family_ids = [family['id'] for family in families]
+
+    tmp_family_tsv = 'tmp_families.tsv'
+    family_tsv_headers = ['Family ID', 'Description', 'Coded Phenotype', 'Display Name']
+    # Work-around as import_families takes a file.
+    with open(tmp_family_tsv, 'wt') as tmp_families:
+        tsv_writer = csv.writer(tmp_families, delimiter='\t')
+        tsv_writer.writerow(family_tsv_headers)
+        for family in families:
+            del family['id']
+            del family['project']
+            tsv_writer.writerow(list(family.values()))
```

Comment:
I'd probably be more explicit with the order you write this in. Trusting the response from the SM server to be in the order with the correct keys is a bit dangerous. I'd probably prefer something that explicitly builds the keys:

```python
tsv_writer.writerow([
    family['external_id'],
    family['description'] or '',
# .... etc
```

Reply:
ah, good point -- thanks! Will fix :)